tiledb/sm/cpp_api/vfs_experimental.h

```diff
@@ -155,6 +192,47 @@ class VFSExperimental {
         &wrapper));
   }
 
+  /**
+   * Recursively lists objects at the input URI, invoking the provided callback
+   * on each entry gathered. The callback is passed the data pointer provided
+   * on each invocation and is responsible for writing the collected results
+   * into this structure. If the callback returns True, the walk will continue.
+   * If False, the walk will stop. If an error is thrown, the walk will stop and
+   * the error will be propagated to the caller using std::throw_with_nested.
+   *
+   * Currently LocalFS, S3, Azure, and GCS are supported.
+   *
+   * @code{.c}
+   * VFSExperimental::LsObjects ls_objects;
+   * VFSExperimental::LsCallbackV2 cb = [&](const std::string_view& path,
+   *                                      uint64_t size,
+   *                                      bool is_dir) {
+   *    ls_objects.emplace_back(path, size);
+   *    return true;  // Continue traversal to next entry.
+   * }
+   *
+   * VFSExperimental::ls_recursive_v2(ctx, vfs, "s3://bucket/foo", cb);
+   * @endcode
+   *
+   * @param ctx The TileDB context.
+   * @param vfs The VFS instance to use.
+   * @param uri The base URI to list results recursively.
+   * @param cb The callback to invoke on each entry.
+   */
+  static void ls_recursive_v2(
```

Comment:
It's not clear to me from the docstring how this is different from `ls_recursive` so that I know which one to use . Applies to the C-API as well

Reply:
> It's not clear to me from the docstring how this is different from ls_recursive

Updated the docs and will bring up the rename in core channel